cx/slack/server.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * TODO(developer): 
+ * Add your service key to the current folder.
+ * Uncomment and fill in these variables.
+ */
+// const slackSigningSecret = '...';
+// const slackToken = 'xoxb-...';
+// const projectId = 'my-project';
+// const locationId = 'global';
+// const agentId = 'my-agent';
+// const languageCode = 'en';
+
+const { WebClient } = require('@slack/web-api');
+const { createEventAdapter } = require('@slack/events-api');
+
+const slackEvents = createEventAdapter(slackSigningSecret);
+const slackClient = new WebClient(slackToken);
+
+// Imports the Google Cloud Some API library
+const {SessionsClient} = require('@google-cloud/dialogflow-cx');
+/**
+ * Example for regional endpoint:
+ *   const location = 'us-central1'
+ *   const client = new SessionsClient({apiEndpoint: 'us-central1-dialogflow.googleapis.com'})
+ */
+const client = new SessionsClient();
+
+/**
+ * Converts Slack request to a detectIntent request.
+ */
+function slackToDetectIntent(slackRequest,sessionPath){
+    const request = {
+        session: sessionPath,
+        queryInput: {
+            text: {
+                text: slackRequest.text,
+            },
+            languageCode,
+        },
+    };
+
+    return request;
+}
+
+/**
+ * Converts detectIntent response to a Slack message request. 
+ */
+function detectIntentToSlackMessage(response,id){
+    var agentResponse = '';
+    
+    for (const message of response.queryResult.responseMessages) {
+        if (message.text) {
+            agentResponse += `${message.text.text}\n`;
+        };
+    };
+    
+    if(agentResponse.length != ''){
+        const request = {
+            channel: id,
+            text: agentResponse
+        };
+        return request;
+    };
+};
+
+/**
+ * Takes as input a request from Slack and converts the request to
+ * detectIntent request which is used to call the detectIntent() function
+ * and finally output the response given by detectIntent().
+ */
+async function detectIntentResponse(slackRequest) {
+    const sessionId = slackRequest.user;
+    const sessionPath = client.projectLocationAgentSessionPath(
+        projectId,
+        locationId,
+        agentId,
+        sessionId
+    );
+    console.info(sessionPath);
+    
+    request = slackToDetectIntent(slackRequest, sessionPath);
+    const [response] = await client.detectIntent(request);
+
+    return response;
+};
+
+slackEvents.on('message', (event) => {
+    if(event.bot_id == '' || event.bot_id == null){
```

Comment:
Is this also working for Slack chat room? I remember there are some special logic to determine if anyone have @ed the bot if it is in chat room

Reply:
Currently it only works when directly talking to the Slack App that can be created from this integration. Although, I am able to add a way for users to ask the Integration app questions in other chat rooms by just mentioning the name of the app in their question. Should I also Include that feature in this integration?

Reply:
It will be great if we can support it in the integration. But we could do it in the next changelist if you want.

Reply:
I will make to sure add the feature in this pull request.

Reply:
The integration is now able to make use of app mentions when in a Slack channal.